addons/l10n_cz/models/account_move_line.py

```diff
@@ -0,0 +1,24 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields, api
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.depends('move_id.taxable_supply_date')
```

Comment:
What If the user changed the accounting date after setting the taxable_supply_date
in this case, I understood that we need to rely on taxable_supply_date not accounting date.
So, Changing the accounting date after setting the taxable_supply_date won't achieve that
Also, Please take a look on this comment https://github.com/odoo/odoo/pull/168933#issuecomment-2222722107

Reply:
The fields in the dependencies act as triggers for the compute method.
Since `move_id.taxable_supply_date` is already a trigger for `move_id.date` and `move_id.date` is in turn a dependency for `currency_rate`, changing either `move_id.date` or `move_id.taxable_supply_date` will trigger the re-computation of the currency rate.
As for the possibility of dependency alteration (which is a valid issue), I don't think we need to worry about it since you wrote a test that would expose it :wink: